From 7b227c7c1ac226498b1ea3987f87350d0f14f49f Mon Sep 17 00:00:00 2001
From: Moritz Haase <Moritz.Haase@bmw.de>
Date: Wed, 2 Jul 2025 09:19:44 +0200
Subject: [PATCH] cmake: Bump required CMake version to 3.5 to allow builds
 with CMake 4+

This enables builds with CMake 4+, fixing:

    CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
      Compatibility with CMake < 3.5 has been removed from CMake.

      Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
      to tell CMake that the project requires at least <min> but has been
      updated to work with policies introduced by <max> or earlier.

      Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

While at it, remove the extra call to 'cmake_policy()'. When passing the same
version as to 'cmake_minimum_required()' it's not needed, as the latter calls
the former automatically.

Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
Upstream-Status: Submitted [https://github.com/WebPlatformForEmbedded/libwpe/pull/136]
---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index db9ae50..2a01daf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
-cmake_policy(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)

 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include(VersioningUtils)
